src/accountsdb/db.zig

```diff
@@ -31,6 +31,7 @@ const AccountIndex = sig.accounts_db.index.AccountIndex;
 const AccountRef = sig.accounts_db.index.AccountRef;
 const RwMux = sig.sync.RwMux;
 const Logger = sig.trace.log.Logger;
+const ScopedLogger = sig.trace.log.ScopedLogger;
```

Comment:
> What do you think? Any downsides?

Applying the suggestion to `GossipDumpService`, it becomes

```zig
pub const GossipDumpService = struct {
    allocator: Allocator,
    logger: Logger,
    gossip_table_rw: *RwMux(GossipTable),
    counter: *Atomic(usize),

    const Self = @This();
    const Logger = sig.trace.ScopedLogger(@typeName(Self));
    ...
```

previously it was

```zig
pub const GossipDumpService = struct {
    allocator: Allocator,
    logger: ScopedLogger(@typeName(Self)),
    gossip_table_rw: *RwMux(GossipTable),
    counter: *Atomic(usize),

    const Self = @This();
```

One minor downside is that it does away with the unwritten convention that `Logger` refers to the upscoped logger, and the scoped logger will be obvious in the type, ie ` logger: ScopedLogger(@typeName(Self)),` in the above.

With the suggestion, the type definition does look cleaner? ie just `Logger`. But there is something to be said about being able to look at the logger and know this is using a scoped logger or not.

Reply:
I feel like you could make the same argument that the `logger` field in a struct should also indicate that it is a scoped logger. For example, why wouldn't you do it like this?
```zig
pub const GossipDumpService = struct {
    allocator: Allocator,
    gossip_dump_service_scoped_logger: ScopedLogger(@typeName(Self)),
    gossip_table_rw: *RwMux(GossipTable),
    counter: *Atomic(usize),

    const Self = @This();
```
It just seems kind of redundant to specify the scope in every variable's name, when the scope itself already is a namespace.

Upon further thought, I think I see your point. I'll answer my own question...

We often need to have the same scope deal with both scoped and unscoped loggers. For example, an `init` function may need to take an unscoped logger, and then convert it to a scoped logger. 

To deal with that you have several options:
1. "Logger" always means unscoped, and use `ScopedLogger("thescope")` anywhere you need a scoped logger.
2. "Logger" always means unscoped, and you assign a new name to the individually scoped logger, like `const ScopedLogger = ScopedLogger("gossip")` (or "GossipLogger" or whatever)
3. "Logger" always means "the correct logger for this scope" and you can specify `sig.trace.Logger` if you need to deal with an unscoped logger in a context where the .
4. "Logger" always refers the most generic logger to use in the current scope. If `sig.trace.Logger` is needed in this scope, you assign it the name "Logger" and use a different name for scoped loggers. If this scope only needs scoped loggers, you can assign the scoped logger to the name "Logger".

I still like option 3 but I do see the potential for confusion. If you prefer options 1 and 2 then that's fine by me. Option 4 feels a bit too chaotic.

Reply:
> so I don't know what's best. If you prefer options 1 and 2 then that's fine by me. 

I do have a slight preference for option 1. Mainly because variable names can be free to be `logger` or whatever makes sense in the context, and a look at the corresponding type gives the extra info (if it is scoped with a non-null value or not)...mirroring how any type inherently offers context for any variable.

> Option 4 feels a bit too chaotic.

Indeed.